File: November2021/C++/Search_Insert_Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int l=0,r=nums.size()-1;
        int mid;
        while(l<=r)
        {
            mid=l+(r-l)/2;
            if(nums[mid]<target)
                l=mid+1;
            else
                r=mid-1;
        }
        return l;
    }
};

File: November2021/SearchInsertPosition.java
// TC : O(logn)
// SC : O(1)
class Solution {
	public int searchInsert(int[] nums, int target) {
		int low = 0;
		int high = nums.length;
		while(low<high){
			int mid = low + (high -low)/2;
			if(nums[mid]>=target){
				high = mid;
			} else {
				low = mid +1;
			}
		}
		return low;
	}
}